sklearn/preprocessing/imputation.py

```diff
@@ -336,16 +336,14 @@ def transform(self, X):
         valid_mask = np.logical_not(invalid_mask)
         valid_statistics = statistics[valid_mask]
         valid_statistics_indexes = np.where(valid_mask)[0]
-        missing = np.arange(X.shape[not self.axis])[invalid_mask]
```

Comment:
Maybe it's not good to change anything in ``Imputer``?

Reply:
The line is not use so I remove it. I don't think this is harmful.

Reply:
@glemaitre Seems that ``missing`` is used in the warning message below (and you also modify the warning message)

Reply:
Oh this is from the previous PR. I will revert those changes then.